table/tables/index.go

```diff
@@ -224,13 +261,39 @@ func (c *index) Delete(sc *stmtctx.StatementContext, txn kv.Transaction, indexed
 	if err != nil {
 		return err
 	}
+
+	key, tempKey, tempKeyVer := genTempIdxKeyByState(c.idxInfo, key)
+
 	if distinct {
-		err = txn.GetMemBuffer().DeleteWithFlags(key, kv.SetNeedLocked)
+		if len(key) > 0 {
+			err = txn.GetMemBuffer().DeleteWithFlags(key, kv.SetNeedLocked)
+			if err != nil {
+				return err
+			}
+		}
+		if len(tempKey) > 0 {
+			val := []byte("deleteu?")
```

Comment:
Please add constants for `deleteu?` and `delete?`.

Reply:
It could reduce the readability if we separate this line from the next line. `?` is a placeholder which will be replaced by `tempKeyVer`.